vignettes/vetiver.Rmd

```diff
@@ -0,0 +1,150 @@
+
+```{r}
+library(parsnip)
+library(recipes)
+library(workflows)
+library(dplyr)
+data(bivariate, package = "modeldata")
+bivariate_train
+
+biv_rec <-
+  recipe(Class ~ ., data = bivariate_train) %>%
+  step_BoxCox(all_predictors())%>%
+  step_normalize(all_predictors())
+
+svm_spec <-
+  svm_linear(mode = "classification") %>%
+  set_engine("LiblineaR")
+
+svm_fit <- 
+  workflow(biv_rec, svm_spec) %>%
+  fit(sample_frac(bivariate_train, 0.7))
+```
+
+This `svm_fit` object is a fitted workflow, with both feature engineering and model parameters estimated using the training data `bivariate_train`. We can create a `vetiver_model()` from this trained model; a `vetiver_model()` collects the information needed to store, version, and deploy a trained model.
+
+```{r}
+library(vetiver)
+v <- vetiver_model(svm_fit, "biv_svm")
+v
+```
+
+Think of this `vetiver_model()` as a deployable model object.
+
+## Store and version your model
+
+You can store and version your model by choosing a [pins](https://pins.rstudio.com) "board" for it, including a local folder, RStudio Connect, Amazon S3, and more. Most pins boards have versioning turned on by default, but we can turn it on explicitly for our temporary demo board. When we write the `vetiver_model()` to our board, the binary model object is stored on our board together with necessary metadata, like the packages needed to make a prediction and the model's input data prototype for checking new data at prediction time.
+
+```{r}
+library(pins)
+model_board <- board_temp(versioned = TRUE)
+model_board %>% vetiver_pin_write(v)
+```
+
+Let's train our model again with a new version of the dataset and write it once more to our board.
+
+```{r}
+svm_fit <- 
+  workflow(biv_rec, svm_spec) %>%
+  fit(sample_frac(bivariate_train, 0.7))
+v <- vetiver_model(svm_fit, "biv_svm")
+
+model_board %>% vetiver_pin_write(v)
+```
+
+Both versions are stored, and we have access to both.
+
+```{r}
+model_board %>% pin_versions("biv_svm")
+```
+
+The primary purpose of pins is to make it easy to share data artifacts, so depending on the board you choose, your pinned `vetiver_model()` can be shareable with your collaborators.
+
+## Deploy your model
+
+You can deploy your model by creating a [Plumber](https://www.rplumber.io/) router, and adding a POST endpoint for making predictions.
+
+```{r}
+library(plumber)
+pr() %>%
+  vetiver_pr_predict(v)
+```
+
+To start a server using this object, next pipe (`%>%`) to `pr_run(port = 8088)` or your port of choice. This allows you to interact with your vetiver API locally and debug it. Plumber APIs such as these can be [hosted in a variety of ways](https://www.rplumber.io/articles/hosting.html). You can use the function `vetiver_write_plumber()` to create a ready-to-go `plumber.R` file that is especially suited for [RStudio Connect](https://www.rstudio.com/products/connect/).
```

Comment:
```suggestion
To start a server using this object, pipe (`%>%`) to `pr_run(port = 8088)` or your port of choice. This allows you to interact with your vetiver API locally and debug it. Plumber APIs such as these can be [hosted in a variety of ways](https://www.rplumber.io/articles/hosting.html). You can use the function `vetiver_write_plumber()` to create a ready-to-go `plumber.R` file that is especially suited for [RStudio Connect](https://www.rstudio.com/products/connect/).
```

This "next" threw me a little and it took longer than necessary to recognise "pipe" as a verb rather than a noun. Might well just be me though :smile: